File: лаба2/Swift/laba2_2zadanie.swift
import Foundation

func main() {
    print("Введите N и M: ")
    guard let input = readLine() else { return }
    let parts = input.components(separatedBy: " ")
    
    guard parts.count == 2 else { print("Неправильный формат ввода"); return }
    
    guard let N = Int(parts[0]), let M = Int(parts[1]) else { print("Неправильный формат ввода"); return }

    var total = 0         // Общее количество выпитых банок
    var emptybot = 0      // Количество пустых банок
    var iterations = 0    // Количество итераций
    var n = N            // Количество полных банок

    print("\nДано:\n")
    for _ in 1...n {
        print("☺", terminator: "")
    }
    print("\n")

    print("\nПроцесс:\n")
    while n > 0 || emptybot >= M { //выпиваем
        total += n
        emptybot += n
        n = 0
        iterations += 1

        for _ in 1...emptybot {
            print("©", terminator: "")
        }
        print(" (пустые банки)\n")

        // обмен
        if emptybot >= M {
            n = emptybot / M // получаем банки
            emptybot %= M // остаток пустых банок
            iterations += 1
            for _ in 1...n {
                print("☺", terminator: "")
            }
            print(" (полные банки после обмена)\n")
        }
    }

    print("\nВыпитые банки: \(total)\n")
    print("Итерации: \(iterations)\n")
}

main()

File: лаба2/Rust/laba2_2zadanie.rs
use std::io;

fn main() {
    println!("Введите M и N: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let mut parts = input.trim().split_whitespace();
    let mut m: i32 = parts.next().unwrap().parse().expect("Please type a number!");
    let mut n: i32 = parts.next().unwrap().parse().expect("Please type a number!");

    let mut total = 0; // Общее количество выпитых банок
    let mut emptybot = 0; // Количество пустых банок
    let mut iterations = 0; // Количество итераций

    println!("\nДано:");
    for _ in 0..n {
        print!("☺");
    }
    println!();

    println!("\nПроцесс:");
    while n > 0 || emptybot >= m {
        total += n;
        emptybot += n;
        n = 0;
        iterations += 1;

        for _ in 0..emptybot {
            print!("©");
        }
        println!(" (пустые банки)");

        // обмен
        if emptybot >= m {
            n = emptybot / m; // получаем банки
            emptybot %= m; // остаток пустых банок
            iterations += 1;
            for _ in 0..n {
                print!("☺");
            }
            println!(" (полные банки после обмена)");
        }
    }

    println!("\nВыпитые банки: {}", total);
    println!("Итерации: {}", iterations);
}

